### PR TITLE
restore: don't restore auto id if the table doesn't has it.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve? 
Fix CI failure after [tidb#18326](https://github.com/pingcap/tidb/pull/18326)

### What is changed and how it works?
We don't try to restore auto ID for tables that don't need it.

### Check List 

Tests 

 - Integration test

### Release Note

 - fix a bug that causes, we will restore auto id even the table doesn't has it.

### CODE REVIEW

**Comment by kennytm** [comment]
The `br_move_backup` test is stuck

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/2308/pipeline/96

File: pkg/restore/db.go [hunk 1]
```diff
@@ -163,9 +164,11 @@ func (db *DB) CreateTable(ctx context.Context, table *utils.Table) error {
 			utils.EncloseName(table.Db.Name.O),
 			utils.EncloseName(table.Info.Name.O),
 			table.Info.AutoIncID)
+		if table.NeedRebaseAutoID() {
```
Comment (kennytm) [inline pkg/restore/db.go:167]: 🤔  why not just use `NeedAutoID(table.Info)` here?

Comment (YuJuncen) [reply pkg/restore/db.go:167]: Because this function isn't written in the first version 🧐.

You'r right...

**Pull request merged**